### PR TITLE
Android: Updated toolchain file and dependencies

### PR DESCRIPTION
**Pull request by MarioLiebisch**
* This PR supercedes the branch *feature/android_toolchain*.
* Replaced the toolchain file with a new version based on [zuhowei's fork](https://github.com/zhuowei/android-cmake), which supports the latest NDK versions as well as 64 bit targets.
* Removed the STL dependency from **sfml-activity** rather than relying on some implementation implicitly linked by default.
* Deleted *project.properties*, which wasn't supposed to be part of the repository code.
* Updated CMake files to properly support Nvidia's Nsight Tegra Visual Studio Edition as an alternative to using the toolchain file (you might still have to update the STL implementation as that's not yet set through CMake).

**Some Considerations:**

* `getLibraryName()` is used only once. As such we could skip the whole string handling and merge this function into `loadLibrary()`.

### CODE REVIEW

**Comment by MarioLiebisch** [comment]
Updated the branch. It's now possible to build SFML without utilizing the toolchain file if you're using Visual Studio and Tegra Nsight VS Edition:

    cmake -G "Visual Studio 12 2013" -DCMAKE_SYSTEM_NAME=Android -DANDROID_NATIVE_API_LEVEL=9 -DANDROID_ABI=armeabi path/to/source

File: cmake/Macros.cmake [hunk 1]
```diff
@@ -176,3 +176,25 @@ macro(sfml_add_example target)
     endif()
 
 endmacro()
+
+# macro to find packages on the host OS
+# this is the same as in the toolchain file, which is here for Nsight
+macro( find_host_package )
```
Comment (eXpl0it3r) [inline cmake/Macros.cmake]: Could you adapt the code styling from above for this and the follow lines?

**Comment by eXpl0it3r** [comment]
> `getLibraryName()` is used only once. As such we could skip the whole string handling and merge this function into `loadLibrary()`.

What are the chances that it will be used multiple times at a later point?
Is the "string handling" an issue or can it be an issue?
Is it a "public" function?

Let me know if you intend to keep working on this PR or rather when it's ready to be merged.

File: cmake/Macros.cmake (hunk 1 above)
Comment (MarioLiebisch) [inline cmake/Macros.cmake]: Hm, yeah, just me doing a quick copy and paste and forgetting about it. :)

**Comment by MarioLiebisch** [comment]
> What are the chances that it will be used multiple times at a later point?

It's private (i.e. even static) and very unlikely to be reused. I can hardly think of any reason to determine the current library name. This is a bit like determining/using `argv[0]` in a desktop program, but even less useful since you can't use it for recursive calls. Only case I could imagine using this would be to more or less DRM a shared library so it can only be used by a specific other shared library, but at the same time the common user will never even see these names.

**Comment by MarioLiebisch** [comment]
I'm going to squash and merge this today unless I hear anything which would prevent this. I'll leave `getLibraryName()` the way it is right now.

**Comment by eXpl0it3r** [comment]
You can squash it, I'll then merge it. :wink:

**Comment by eXpl0it3r** [comment]
I wouldn't add the `examples/android/.gitignore` file. It doesn't really serve any purpose.

**Comment by MarioLiebisch** [comment]
Well, it keeps git's status clean when building in-tree (which you typically do), but then again it's not necessary... hm...

**Comment by eXpl0it3r** [comment]
It's never really recommended to build in tree and if you want the status to be clean we'd have to add a ton of additional git ignores, which is just ugly and doesn't really help.

**Comment by LaurentGomila** [comment]
> when building in-tree (which you typically do)

Nooooooooooooooooo. Nobody does that. Not in a source tree that you're constantly working in, at least. Please make you a favor, and create your build directory out of the source tree :stuck_out_tongue:

**Comment by MarioLiebisch** [comment]
I obviously do. :D But okay, point taken, removing it.

**Comment by MarioLiebisch** [comment]
There you go, squashed, rebased and ready. :)

**Comment by eXpl0it3r** [comment]
This PR has been added to my merge list, meaning it will be merged soon, unless someone raises any concerns.

**Comment by LaurentGomila** [comment]
There are tabs all over the place in CMake files, uppercase functions ("SET" instead of "set"), and what's the thing with Nsight?

**Comment by MarioLiebisch** [comment]
Gna, there shouldn't be tabs~ Nsight is a plugin for Visual Studio providing NDK integration (compiling/linking, debugging, etc.).

**Comment by MarioLiebisch** [comment]
Okay, updated the branch - I didn't touch the new toolchain file though, since I'd like to keep it 1:1 to the original (linked above).

**Comment by eXpl0it3r** [comment]
If it's an "external dependency" you should exlude it from the .gitattributes (and .editorconfig).

**Comment by MarioLiebisch** [comment]
Added. EditorConfig doesn't support exclusions as far as I know, but should be fine (since the shouldn't be edited anyway; similar to stb stuff).

**Comment by LaurentGomila** [comment]
> Nsight is a plugin for Visual Studio providing NDK integration 

Ok but my question was more: why does it need special stuff (find_host_package) in addition to the regular toolchain?

**Comment by eXpl0it3r** [comment]
The latest change on master has created a conflict with this branch. Make sure to rebase and resolve.

**Comment by MarioLiebisch** [comment]
> Ok but my question was more: why does it need special stuff (find_host_package) in addition to the regular toolchain?
Simply because it doesn't use the toolchain at all. It's a "special" target built into CMake, which is triggered by setting `CMAKE_SYSTEM_NAME` to `Android` while creating a MSVC project.

Going to add a short note together with the rebase/merge.

**Comment by MarioLiebisch** [comment]
Done.

**Comment by eXpl0it3r** [comment]
This PR has been added to my merge list, meaning it will be merged soon, unless someone raises any concerns.

**Comment by eXpl0it3r** [comment]
Building this for testing I noticed that you added (at least) three options, but didn't mark them with the `SFML_` prefix.
Also is it possible to only "display"/set them when one is actually trying to build for Android, otherwise it can be a bit irritating.

Android

**Comment by MarioLiebisch** [comment]
They shouldn't appear when you're not building for Android. Do they still appear for you? If so, that's a bug. Unfortunately they have to be set before the toolchain is invoked (using CMake's `project()` command), so there's no way in using your usual CMake variables for branching.

We could wrap those with a `SFML_` prefix, but behind the scenes we'd still have to set the non-prefixed variables. Those are used by the toolchain file and CMake.

**Comment by MarioLiebisch** [comment]
I've updated the branch. Try again, they should be gone now. This still leaves the name thing though.

**Comment by eXpl0it3r** [comment]
Ah I guess the naming is okay if it's used by the toolchain. I was mostly confused to see the options on my standard MinGW build.

Seems to work fine now. :+1:

**Pull request merged**